### DISCUSSION TITLE
get the libnode.dll

### DISCUSSION THREAD

**Discussion by kokashking** [microsoft/node-api-dotnet#299] [Q&A]
Hello, community!

I want to build the C# WinUI + JS Fluid Framework example. (https://github.com/microsoft/node-api-dotnet/tree/main/examples/winui-fluid)

For that I need the libnode.dll. Can you please provide me the library?

Thank you in advance!

**Comment by jasongin** [comment]
`libnode` binaries are not published yet but you can build your own. See https://github.com/microsoft/node-api-dotnet/issues/107#issuecomment-2057928923

**Reply by kokashking** [reply]
@jasongin  Thank you for the answer. I have built the libnode.dll. 

To my regret when I try to run the winui-fluid example in the Packaged mode its not possible to deploy because I am missing:
winui-fluid\bin\x64\Debug\net8.0-windows10.0.22621.00\win-x64\winui-fluid.build.appxrecipe

And when I try to run the Unpackaged version I get the exception on line 30, in the Constructor _new NodejsPlatform(libnodePath)_ :
_System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'_

**Reply by jasongin** [reply]
> I am missing:
winui-fluid\bin\x64\Debug\net8.0-windows10.0.22621.00\win-x64\winui-fluid.build.appxrecipe

Sorry I am not an expert in winui app deployment so I don't know how to solve that.

That example project code had gotten a bit stale so it needed some small fixes, which I did just now: https://github.com/microsoft/node-api-dotnet/pull/311

After that PR, it worked for me, running unpackaged. I don't know what caused the AV you saw.

**Comment by jasongin** [comment]
Update: Instructions for building libnode are at https://microsoft.github.io/node-api-dotnet/scenarios/dotnet-js.html
